File: 10_001st_Prime/Golang/main.go
package main

import "fmt"

func main() {
	var res []int = sieve_of_eratosthenes(100000000)
	fmt.Print(res[10000])
}

func sieve_of_eratosthenes(n int) []int{
	sieve := make([]bool,n+1)
	var primes []int
	for i := range sieve{
		sieve[i] = true
	}

	for p := 2; p * p <= n; p++{
		if sieve[p]{
			for i := p * p; i <= n; i += p{
				sieve[i] = false
			}
		}
	}
	
	for i, v := range sieve{
		if v{
			primes = append(primes, i)
		}
	}
	return primes[2:]
}

File: 10_001st_Prime/Python/main.py
def sieve_of_eratosthenes(n:int):
    sieve = [True] * (n + 1)
    primes = []
    p = 2
    while p * p <= n:
        if sieve[p]:
            for i in range(p * p,n + 1,p):
                sieve[i] = False
        p += 1
    
    for i, v in enumerate(sieve):
        if v:
            primes.append(i)
    return primes[2:]

print(sieve_of_eratosthenes(100000000)[10000])